examples/tutorial/visuals/01_basic_visual.py

```diff
@@ -0,0 +1,165 @@
+# vispy: testskip
+"""
+Tutorial: Creating Visuals
+--------------------------
+
+This tutorial is intended to guide developers who are interested in creating 
+new subclasses of Visual. In most cases, this will not be necessary because
+vispy's base library of visuals will be sufficient to create complex scenes as
+needed. However, there are cases where a particular visual effect is desired 
+that is not supported in the base library, or when a custom visual is needed to
+optimize performance for a specific use case.
+
+The purpose of a Visual is to encapsulate a single drawable object. This
+drawable can be as simple or complex as desired. Some of the simplest visuals 
+draw points, lines, or triangles, whereas more complex visuals invove multiple
+drawing stages or make use of sub-visuals to construct larger objects.
+
+In this example we will create a very simple Visual that draws a rectangle.
+Visuals are defined by:
+
+1. Creating a subclass of vispy.visuals.Visual
+2. Defining a draw() method that takes into account some user-specified
+   transformation functions.
+
+
+"""
+from vispy import app, gloo, visuals, scene
+import numpy as np
+
+# Define a simple vertex shader. We use $template variables as placeholders for
+# code that will be inserted later on. In this example, $position will become
+# an attribute, and $transform will become a function. Important: using
+# $transform in this way ensures that users of this visual will be able to 
+# apply arbitrary transformations to it.
+vertex_shader = """
+void main() {
+   gl_Position = $transform(vec4($position, 0, 1));
+}
+"""
+
+# Very simple fragment shader. Again we use a template variable "$color", which
+# allows us to decide later how the color should be defined (in this case, we
+# will just use a uniform red color).
+fragment_shader = """
+void main() {
+  gl_FragColor = $color;
+}
+"""
+
+
+# Start the new Visual class. 
+# By convention, all Visual subclass names and in 'Visual'
```

Comment:
typo "and" -> "end". Also probably worth mentioning that they MUST end in Visual or else `create_visual_node()` won't work